### PR TITLE
Add support to LCD ST7789 on STM32F4Discovery

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Modify the ST7789 driver to allow change X/Y resolution
and
Add support to LCD ST7789 on STM32F4Discovery
## Impact
Only on ST7789 and STM32F4Discovery
## Testing
Tested on STM32F4Discovery

### CODE REVIEW

File: boards/arm/stm32/stm32f4discovery/src/stm32_spi.c [hunk 1]
```diff
@@ -133,9 +134,11 @@ void weak_function stm32_spidev_initialize(void)
  ****************************************************************************/
 
 #ifdef CONFIG_STM32_SPI1
-void stm32_spi1select(FAR struct spi_dev_s *dev, uint32_t devid, bool selected)
+void stm32_spi1select(FAR struct spi_dev_s *dev, uint32_t devid,
+                      bool selected)
 {
-  spiinfo("devid: %d CS: %s\n", (int)devid, selected ? "assert" : "de-assert");
+  spiinfo("devid: %d CS: %s\n", (int)devid, selected ? "assert" : \
```
Comment (xiaoxiang781216) [inline boards/arm/stm32/stm32f4discovery/src/stm32_spi.c:140]:
```suggestion
  spiinfo("devid: %d CS: %s\n", (int)devid, selected ? "assert" :
```

File: boards/arm/stm32/stm32f4discovery/src/stm32_spi.c [hunk 2]
```diff
@@ -196,9 +199,11 @@ uint8_t stm32_spi1status(FAR struct spi_dev_s *dev, uint32_t devid)
 #endif
 
 #ifdef CONFIG_STM32_SPI2
-void stm32_spi2select(FAR struct spi_dev_s *dev, uint32_t devid, bool selected)
+void stm32_spi2select(FAR struct spi_dev_s *dev, uint32_t devid,
+                      bool selected)
 {
-  spiinfo("devid: %d CS: %s\n", (int)devid, selected ? "assert" : "de-assert");
+  spiinfo("devid: %d CS: %s\n", (int)devid, selected ? "assert" : \
```
Comment (xiaoxiang781216) [inline boards/arm/stm32/stm32f4discovery/src/stm32_spi.c:205]: ditto

File: boards/arm/stm32/stm32f4discovery/src/stm32_spi.c [hunk 3]
```diff
@@ -239,9 +244,11 @@ uint8_t stm32_spi2status(FAR struct spi_dev_s *dev, uint32_t devid)
 #endif
 
 #ifdef CONFIG_STM32_SPI3
-void stm32_spi3select(FAR struct spi_dev_s *dev, uint32_t devid, bool selected)
+void stm32_spi3select(FAR struct spi_dev_s *dev, uint32_t devid,
+                      bool selected)
 {
-  spiinfo("devid: %d CS: %s\n", (int)devid, selected ? "assert" : "de-assert");
+  spiinfo("devid: %d CS: %s\n", (int)devid, selected ? "assert" : \
```
Comment (xiaoxiang781216) [inline boards/arm/stm32/stm32f4discovery/src/stm32_spi.c:250]: ditto

**Comment by acassis** [comment]
Thank you @xiaoxiang781216 fixed and squashed!

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**